src/Compiler/Checking/AugmentWithHashCompare.fs

```diff
@@ -1351,13 +1351,13 @@ let MakeBindingsForCompareAugmentation g (tycon: Tycon) =
                 let tinst, ty = mkMinimalTy g tcref
 
                 let thisv, thise = mkThisVar g m ty
-                let thatobjv, thatobje = mkCompGenLocal m "obj" g.obj_ty
+                let thatobjv, thatobje = mkCompGenLocal m "obj" g.obj_ty_ambivalent
```

Comment:
is this null ambivalent?

Reply:
This is an inner value which is not exposed.
ambivalent => we do not care.